### PR TITLE
Add ZipCode Controller and Tests

### LINKED ISSUE #14: Add Zipcode Controller and Tests
# Acceptance Criteria:

- [ ] There is <tt>ZipcodeController.java</tt> file in the `src/main/java/edu/ucsb/cs156/spring/backend/controllers` directory.
- [ ] <tt>ZipcodeController.java</tt> implements a class that implements an API endpoint that takes in the necessary
      parameters, calls the <tt>ZipcodeQueryService</tt>, and returns the results.
- [ ] The file  <tt>ZipcodeController.java</tt> file make appropriate use of the 
      annotations `@Tag`, `@Operation` and `@Parameter` to document the API.
- [ ] The controller endpoint shows up on the Swagger UI page.
- [ ] There is a <tt>ZipcodeControllerTests.java</tt> file in the `src/test/java/edu/ucsb/cs156/spring/backend/controllers` directory.
- [ ] The tests in <tt>ZipcodeControllerTests</tt> provide 100% coverage of <tt>ZipcodeController</tt> for both jacoco and pitest.
      
# Details

See the assignment description at https://ucsb-cs156.github.io/m23/lab/team01 for details.

### PR DESCRIPTION
**Pull request by Edwin-Yee**
In this PR, I implement a ZipCode service and ZipCode controller using the API from this site https://api.zippopotam.us/. The endpoint `/api/zipcode/get` can be used to get information about the location of a particular zip code.  

Dev Deployment: http://team01-edwin-yee-dev.dokku-06.cs.ucsb.edu/

image

Closes #14  Add Zipcode Controller and Tests

### CODE REVIEW

File: src/main/java/edu/ucsb/cs156/spring/backenddemo/controllers/HomeController.java [hunk 1]
```diff
@@ -43,10 +43,12 @@ public static String getHomePageObjectJSON(String baseUrl) throws JsonProcessing
         resultMap.put("greeting","Greetings from Spring Boot!");
```
Comment (mier2) [inline src/main/java/edu/ucsb/cs156/spring/backenddemo/controllers/HomeController.java:43]: Thank you for your work, I appreciate it. I am wondering maybe if you can submit separate pull requests for different issues. Like here you  are addressing issue 14, but you also modified file for issue 3. Maybe we can all work on issue 3 together when everyone is done with their assigned work.

Comment (mier2) [reply src/main/java/edu/ucsb/cs156/spring/backenddemo/controllers/HomeController.java:43]: Also maybe you can try to look through your code again to see if you can pass those automated pass before we proceed furthur with code review and merging this pull request

**Review by mier2 — CHANGES_REQUESTED** [review]

File: src/main/java/edu/ucsb/cs156/spring/backenddemo/controllers/HomeController.java (hunk 1 above)
Comment (Edwin-Yee) [reply src/main/java/edu/ucsb/cs156/spring/backenddemo/controllers/HomeController.java]: I updated the code so that I do not modify the file for issue 3. The automated checks now pass.

**Review by ryan-he-8 — APPROVED** [review]
lgtm
